File: medxc-gui/src/app/model/patient.ts
import { Appointment } from './appointment';

export class Patient {
    id: string;
    name: string;
    email: string;
    sex: string;
    phoneNumber: string;
    address: string;
    appointments: Appointment[];
    constructor(
        id: string,
        name: string,
        email: string,
        sex: string,
        phoneNumber: string,
        address: string,
        appointments: Appointment[],
    ) {
        this.id = id;
        this.name = name;
        this.email = email;
        this.sex = sex;
        this.phoneNumber = phoneNumber;
        this.address = address;
        this.appointments = appointments;
    }
}

File: medxc-persistence/src/main/java/com/dxc/medxc/persistence/model/Patient.java
/**
 * 
 */
package com.dxc.medxc.persistence.model;

import java.util.List;

/**
 * @author amirchev
 */
public interface Patient {

    /**
     * @return PIN - id of patient
     */
    String getPIN();

    /**
     * @return email of patient
     */
    String getEmail();

    /**
     * @return name of patient
     */
    String getName();

    /**
     * @return patient's phone number
     */
    String getPhoneNumber();

    /**
     * @return patient's sex
     */
    char getSex();

    /**
     * @return patient's address
     */
    String getAddress();
    
    /**
     * @return list of patient's appointments
     */
    List<Appointment> getAppointments();
}
